new-components/reporters/slack/internal/reporter/reporter.go

```diff
@@ -0,0 +1,129 @@
+package reporter
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/go-errors/errors"
+
+	"github.com/smithy-security/pkg/env"
+
+	"github.com/smithy-security/smithy/sdk/component"
+	vf "github.com/smithy-security/smithy/sdk/component/vulnerability-finding"
+)
+
+type Conf struct {
+	SlackWebhook string
+}
+
+type slackLogger struct {
+	Conf *Conf
+}
+
+// NewSlackLogger returns a new slack logger.
+func NewSlackLogger(c *Conf) slackLogger {
+	return slackLogger{
+		Conf: c,
+	}
+}
+
+// NewConf returns a new configuration build from environment lookup.
+func NewConf(envLoader env.Loader) (*Conf, error) {
+	var envOpts = make([]env.ParseOption, 0)
+	if envLoader != nil {
+		envOpts = append(envOpts, env.WithLoader(envLoader))
+	}
+
+	webhook, err := env.GetOrDefault(
+		"SLACK_WEBHOOK",
+		"",
+		append(envOpts, env.WithDefaultOnError(false))...,
+	)
+	if err != nil {
+		return nil, errors.Errorf("could not get env variable for SLACK_WEBHOOK: %w", err)
+	}
+	if webhook == "" {
```

Comment:
I think this is not necessary as `env.WithDefaultOnError` is set to false

Reply:
not sure what you mean here

Reply:
I mean that the check for the empty webhook is superfluous because `env.WithDefaultOnError(false)` already returns an error when the value is empty :)